### PR TITLE
test: Add tests for cli.py

### LINKED ISSUE #635: add tests for the cli(__main__.py)
we will have unit test the cli. Intergration tests there will be very flaky.

### PR DESCRIPTION
**Pull request by Mr-Sunglasses**
**Description**

This PR fixes #635

### CODE REVIEW

**Comment by Mr-Sunglasses** [comment]
> **Description**
> 
> This PR fixes #635

Hey 👋🏻 @sansyrox , back again to Robyn after a long time, Looking to picking up more issues 🥂

File: unit_tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        assert mock_open.call_args[0][0] == "https://robyn.tech"
+
+
+@patch("robyn.reloader.setup_reloader")
+@patch("os.environ", {"IS_RELOADER_RUNNING": "True"})
+def test_start_dev_server_already_running(mock_setup):
+    config = Config()
+    start_dev_server(config)
+    mock_setup.assert_not_called()
+
```
Comment (sansyrox) [inline unit_tests/test_cli.py:67]: Could you please make it a fixture and move it to the `unit_tests/conftest.py` file?

File: unit_tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    else:
+        mock_remove.assert_not_called()
+
+
+def test_docs():
+    with patch("webbrowser.open") as mock_open:
+        docs()
+        mock_open.assert_called_once_with("https://robyn.tech")
+        assert mock_open.call_args[0][0] == "https://robyn.tech"
+
```
Comment (sansyrox) [inline unit_tests/test_cli.py:59]: Could you move it to a separate file?

File: unit_tests/test_create_robyn_app.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import pytest
+import os
+from unittest.mock import patch
+from robyn.cli import create_robyn_app, SCAFFOLD_DIR, CURRENT_WORKING_DIR
+
+
+@pytest.fixture
+def project_directory():
+    return CURRENT_WORKING_DIR / "test_dir"
```
Comment (sansyrox) [inline unit_tests/test_create_robyn_app.py:9]: Hey @Mr-Sunglasses 👋 

The windows tests are failing due to this path joining. Try doing `Path.resolve` here

File: unit_tests/test_create_robyn_app.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        "docker": docker_input,
+        "project_type": project_type,
+    }
+    create_robyn_app()
+    expected_project_dir = project_directory.resolve()
+    mock_makedirs.assert_called_with(expected_project_dir, exist_ok=True)
+    expected_template_path = (SCAFFOLD_DIR / project_type).resolve()
+    mock_copytree.assert_called_with(str(expected_template_path), str(expected_project_dir), dirs_exist_ok=True)
+    if should_remove_dockerfile:
+        expected_dockerfile_path = os.path.join(expected_project_dir, "Dockerfile")
```
Comment (sansyrox) [inline unit_tests/test_create_robyn_app.py:49]: Let use Path from pathlib here https://docs.python.org/3/library/pathlib.html#basic-use

File: unit_tests/test_create_robyn_app.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        mock_remove.assert_called_once_with(expected_dockerfile_path)
```
Comment (sansyrox) [inline unit_tests/test_create_robyn_app.py:50]: this should fix the error here

**Comment by Mr-Sunglasses** [comment]
Hey 👋🏻 @sansyrox , I think I need to revisit the `cli.py` implementation and check weather it is working in windows or not as using pathlib does not resolve this issue in actions 😢 .

**Comment by sansyrox** [comment]
Hey @Mr-Sunglasses , closing this as the tests have been implemented now.

**Pull request closed without merging**